### PR TITLE
Gracefully handle disconnection and reconnection of the RFXtrx, witho…

### PR DESCRIPTION
**Pull request by maxwellhadley**
…ut throwing errors

### CODE REVIEW

**Comment by maxwellhadley** [comment]
Sorry for the extra commit  - finger trouble on my part! This PR improves the serialport handling so that you can unplug the RFXtrx433 and plug it back in again without throwing errors or crashing the Node runtime. By handling the new extra events appropriately, you can automatically reconnect to the RFXtrx433 when (if!) it is reconnected. I've put in code which enforces the recommended 5 second time-out after the RFXtrx433 powers up during which it will enter the bootloader if it receives any data, and this seems to have finally cured the 'unknown packet type 0' errors we saw last year. There are changes to rfxcom.js & its test script, and I've updated README to explain the new API. I've left the version numbering change up to you - this should be a fully backwards compatible API change, but it does add new functionality, so maybe 0.9.0?

**Comment by maxwellhadley** [comment]
Bump!

File: lib/rfxcom.js [hunk 1]
```diff
@@ -66,50 +103,94 @@ util.inherits(RfxCom, EventEmitter);
 RfxCom.prototype.open = function() {
     var self = this;
 
-    if (typeof self.serialport === "undefined") {
-        self.serialport = new serialport.SerialPort(self.device, {
-            baudrate: 38400,
-            parser: self.rfxtrxParser()
+    if (typeof self.serialport === "undefined" || self.fakingSerialPort) {
```
Comment (bigkevmcd) [inline lib/rfxcom.js]: This doesn't look right, it shouldn't really care that it's a fake serial port?

I do wonder about simplifying it by requiring that you pass a SerialPort into the RfxCom constructor?

Comment (maxwellhadley) [inline lib/rfxcom.js]: The basic mechanics - which is just how you designed it - is that in normal use, the RfxCom object creates a serialport based on the path in the first constructor argument, but in testing use the port property of the second argument is a ready made FakeSerialPort. This probably means that you are right, and the condition at new line 106 is always true, though I haven't yet tested this. I'll look into it, although that last 'Bump!" comment unleashed upon me a whole deluge of 5-minute jobs which had been lurking waiting for me to think I had some spare time...

If this is the case, then if the FakeSerialPort had an open() method (which doesn't need to do anything) we could lose the fakingSerialPort property entirely, which would be nice.

I found that, if the USB cable gets disconnected, after emitting the 'error' or 'disconnect' event, the serialport object is unable to recover or be re-opened when the cable gets reconnected. The disconnectHandler calls RfxCom.close(), which closes the serialport (if needs closing, i.e. it still has a non-falsy file descriptor) and then destroys it. A fresh new serialport is created on the next call to RfxCom.open().

This means that you can't easily pass in a serialport object to the RfxCom constructor, at least not if you treat it as an opaque object. You could retrieve its path name, and use that to recreate the serialport if necessary after a disconnect, but in that case it seems simpler just to pass in the path name as now, and let RfxCom handle the actual Serialport object creation & destruction.

**Comment by maxwellhadley** [comment]
While testing whether fakingSerialPort can be safely removed, I found a corner case that provoked a crashing bug

**Pull request merged**